arangod/Pregel/Actor/include/Actor/Actor.h

```diff
@@ -0,0 +1,223 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <Inspection/VPackWithErrorT.h>
+#include <memory>
+#include <iostream>
+#include <string_view>
+#include <type_traits>
+
+#include "ActorBase.h"
+#include "HandlerBase.h"
+#include "Message.h"
+#include "MPSCQueue.h"
+
+namespace arangodb::pregel::actor {
+
+namespace {
+template<typename Runtime, typename A>
+concept IncludesAllActorRelevantTypes =
+    std::is_class<typename A::State>::value &&
+    std::is_class<typename A::template Handler<Runtime>>::value &&
+    std::is_class<typename A::Message>::value;
+template<typename A>
+concept IncludesName = requires() {
+  { A::typeName() } -> std::convertible_to<std::string_view>;
+};
+template<typename Runtime, typename A>
+concept HandlerInheritsFromBaseHandler =
```

Comment:
Do we use concept? I'm not sure all our compilers good with it

Reply:
Jenkins ran through for this PR so I guess it is fine for all tested compilers. At least in Replication2 a concept is used as well, it improves the error message a lot.